dashboard-app/src/hooks/useRequest.ts

```diff
@@ -1,16 +1,34 @@
 import { useEffect, useState } from 'react';
 import { AxiosPromise, AxiosResponse, AxiosRequestConfig } from 'axios';
-import { Response } from '../response';
+import { Response } from '../util/response';
+import { redirectOnError } from '../util/routing';
 
 type RequestParams = {
   apiCall: (params?: Pick<AxiosRequestConfig, 'params'>) => AxiosPromise
   params?: any
   updateOn?: any[]
   callback?: (d: any) => void
+  setErrors: (d: any) => void
+  push: any
 };
 
-export default ({ apiCall, params = {}, updateOn = [], callback }: RequestParams) => {
-  const [error, setError] = useState<string>();
+const onError = (error: any, setErrors: any, push: any) => {
```

Comment:
The types aren't that involved here, we may as well stick them in:
`(error: AxiosError, setErrors: (d: Dictionary<string>) => void, push: (p: string) => void)`

Reply:
`AxiosError` wasn't playing nicely with the Response utils, as it doesn't have all the same properties as `AxiosResponse`. I've left this one for now.